.github/workflows/python-poetry-release.yaml

```
@@ -0,0 +1,98 @@
+name: Reusable workflow for building and publishing python packages to TestPyPI or PyPI
+
+on:
+  workflow_call:
+    inputs:
+      release-type:
+        description: "Scope of the release; See: https://python-poetry.org/docs/cli/#version"
+        required: true
+        type: string
+      ref:
+        description: "The ref name to checkout the repository."
+        required: false
+        default: ${{ github.event.repository.default_branch }}
+        type: string
+      python-version:
+        description: "The python version for setting up poetry. (Default is 3.7)"
+        required: false
+        default: "3.7"
+        type: string
+      poetry-version:
+        description: "The python version for setting up poetry. (Default is 1.1.12)"
+        required: false
+        default: "1.1.12"
+        type: string
+      publish-to-test:
+        description: "If set to false, the packages are published to pypi.org. (Default is true)"
```

nitpick: it's basically fine as it is
```suggestion
        description: "If set to false, the package is published to PyPI. (Default is true)"
```